ZFSin/zfs/module/zfs/zvol.c

```diff
@@ -2474,7 +2475,8 @@ zvol_read_win(zvol_state_t *zv, uint64_t position,
 
 	lr = rangelock_enter(&zv->zv_rangelock, position, count,
 	    RL_READER);
-	while (count > 0 && (position+offset) < volsize) {
+	request_cancellation = InterlockedCompareExchange((LONG volatile *)&zv->zv_request_cancellation, 0, 0);
```

Comment:
But it gets a bit awkward to use `InterlockedCompareExchange()` in the generic `zvol.c`. In the shared sources, the illumos APIs should probably be used, ie, `atomic_cas_32()`. / `atomic_cas_64`.
Here and 3 below...

Reply:
I agree with Jorgen.  Even though the code is windows specific it makes sense to continue using the atomic_xxx calls and eventually modify their content to suit windows if necessary.